Infrastructure-Scripts/Set-AppRoleAssignments/Set-AppRoleAssignments.ps1

```diff
@@ -0,0 +1,142 @@
+        throw "No app registrations to process. Check app service name or update configuration."
+    }
+
+    foreach ($AppRegistration in $AppRegistrationsToProcess) {
+        $AppRegistrationName = "$ResourceNamePrefix$($AppRegistration.appRegistrationSuffix)"
+        $IdentifierUri = "https://$Tenant/$AppRegistrationName"
+        $ServicePrincipal = Get-ServicePrincipal -DisplayName $AppRegistrationName
+
+        if ($ServicePrincipal.Count -eq 1) {
+            Write-Output "-> Processing app registration $AppRegistrationName"
+        }
+        elseif ($ServicePrincipal.Count -gt 1) {
+            throw "-> Found a duplicate app registrations with the same display name. Investigate"
+        }
+        else {
+            Write-Output "-> App registration $AppRegistrationName not found in AAD - Creating"
+
+            if (!$DryRun) {
+                New-AppRegistration -AppRegistrationName $AppRegistrationName -IdentifierUri $IdentifierUri
+            }
+
+            Write-Output "  -> Successfully created app registration - $AppRegistrationName"
+            $ServicePrincipal = Get-ServicePrincipal -DisplayName $AppRegistrationName
+        }
+
+        foreach ($AppRole in $AppRegistration.appRoles) {
+            $MatchedAppRole = $ServicePrincipal.appRoles | Where-Object { $_.value -eq $AppRole.appRoleName }
+
+            if ($MatchedAppRole) {
+                Write-Output "  -> Processing app role $($AppRole.appRoleName)"
+            }
+            else {
+                Write-Output "  -> App role $($AppRole.appRoleName) not found on app registration - Creating App Role"
+
+                if (!$DryRun) {
+                    New-AppRegistrationAppRole -AppRoleName $AppRole.appRoleName -IdentifierUri $IdentifierUri
+                }
+
+                Write-Output "    -> Successfully added app role $($AppRole.appRoleName)"
+
+                $ServicePrincipal = Get-ServicePrincipal -DisplayName $AppRegistrationName
+                $MatchedAppRole = $ServicePrincipal.appRoles | Where-Object { $_.value -eq $AppRole.appRoleName }
+            }
+            foreach ($MIResource in $AppRole.resourceNameSuffix) {
+                $ManagedIdentityResourceName = "$ResourceNamePrefix$MIResource"
+                $ManagedIdentity = Get-ServicePrincipal -DisplayName $ManagedIdentityResourceName
+                if ($ManagedIdentity.Count -eq 1) {
+                    Write-Output "    -> Processing Managed Identity of $ManagedIdentityResourceName"
+                }
+                elseif ($ManagedIdentity.Count -gt 1) {
+                    Write-Output "    -> Found duplicate app registrations with name $ManagedIdentityResourceName - Investigate"
+                    continue
+                }
+                else {
+                    Write-Output "    -> Managed Identity of $ManagedIdentityResourceName not found"
+                    continue
+                }
+
+                try {
```

Comment:
Do we want this try catch block if its just throwing an error to the other try catch block?

Reply:
the catch in the try block doesnt actually throw. It only writes an error and will continue applying other app role assignments if one fails